### DISCUSSION TITLE
Is load testing of custom gRPC server possible?

### DISCUSSION THREAD

**Discussion by santhoshk** [fortio/fortio#781] [Q&A] [answered]
The question is about using fortio for gRPC load testing..

From the examples in the readme section, I see that we need to first start fortio server and then make ping/health calls to this server.
But in my scenario, I already have a gRPC server (lets say running at http://localhost:5041) and then i need to make calls to the methods exposed by this server, e.g /sayHello with json input 
{
  "name": "TestMessage"
}

The intent is to try to use fortio for load testing my own gRPC servers.
Is it possible to use fortio for the scenario described above. Please point me to some examples on how to do this?

**Comment by ldemailly** [comment, marked as answer]
Sorry for the late reply, github isn't notifying me of new discussions until I actually go and check.

Design wise Fortio was mostly conceived initially to test "in the middle" overhead of istio/envoy proxies and ingress, so its GRPC testing client is mostly about testing its own server on the other side - or the std health check endpoint

There are other tools in go that do dynamic grpc and maybe some day I'll have time to add that but it's not looking likely

This is basically https://github.com/fortio/fortio/issues/420
